crypto/src/lib.rs

```diff
@@ -494,21 +496,22 @@ impl Decode for PublicKey {
     }
 }
 
-/// Private Key used in signatures.
-#[derive(DebugCustom, Display, Clone, PartialEq, Eq, Serialize)]
-#[cfg_attr(feature = "ffi", derive(IntoFfi, TryFromFfi))]
-#[debug(fmt = "{{ digest: {digest_function}, payload: {:X?}}}", payload)]
-#[display(fmt = "{}", "hex::encode(payload)")]
-#[allow(clippy::multiple_inherent_impl)]
-pub struct PrivateKey {
-    /// Digest function
-    digest_function: ConstString,
-    /// key payload. WARNING! Do not use `"string".as_bytes()` to obtain the key.
-    #[serde(with = "hex::serde")]
-    payload: Vec<u8>,
+ffi! {
+    /// Private Key used in signatures.
+    #[derive(DebugCustom, Clone, PartialEq, Eq, Display, Serialize, IntoFfi, TryFromReprC)]
+    #[debug(fmt = "{{ digest: {digest_function}, payload: {:X?}}}", payload)]
+    #[display(fmt = "{}", "hex::encode(payload)")]
+    #[allow(clippy::multiple_inherent_impl)]
+    pub struct PrivateKey {
+        /// Digest function
+        digest_function: ConstString,
+        /// key payload. WARNING! Do not use `"string".as_bytes()` to obtain the key.
+        #[serde(with = "hex::serde")]
+        payload: Vec<u8>,
```

Comment:
We might be able to reduce the size of this as well. 

The `payload` is effectively fixed size. So what we can have is a generic wrapped by a `union`.

Reply:
very true, and worth doing. I'll open an issue

Reply:
#2573